frontend/src/features/admin-form/template/TemplateFormProvider.tsx

```diff
@@ -0,0 +1,187 @@
+import simplur from 'simplur'
+
+import { FormAuthType, FormResponseMode } from '~shared/types/form'
+
+import { useFormTemplate } from '~/features/admin-form/common/queries'
+import { FormNotFound } from '~/features/public-form/components/FormNotFound'
+import {
+  PublicFormContext,
+  SubmissionData,
+} from '~/features/public-form/PublicFormContext'
+import { useCommonFormProvider } from '~/features/public-form/PublicFormProvider'
+
+import { useTimeout } from '~hooks/useTimeout'
+import { HttpError } from '~services/ApiService'
+import { FormFieldValues } from '~templates/Field'
+
+import NotFoundErrorPage from '~pages/NotFoundError'
+
+import { usePreviewFormMutations } from '../common/mutations'
+
+interface PreviewFormProviderProps {
+  formId: string
+  children: React.ReactNode
+}
+
+export const TemplateFormProvider = ({
+  formId,
+  children,
+}: PreviewFormProviderProps): JSX.Element => {
+  // Once form has been submitted, submission data will be set here.
+  const [submissionData, setSubmissionData] = useState<SubmissionData>()
+
+  const { data, isLoading, error, ...rest } = useFormTemplate(
+    formId,
+    // Stop querying once submissionData is present.
+    /* enabled= */ !submissionData,
+  )
+
+  const { isNotFormId, toast, vfnToastIdRef, expiryInMs, ...commonFormValues } =
+    useCommonFormProvider(formId)
+
+  const showErrorToast = useCallback(
+    (error) => {
+      toast({
+        status: 'danger',
+        description:
+          error instanceof Error
+            ? error.message
+            : 'An error occurred whilst processing your submission. Please refresh and try again.',
+      })
+    },
+    [toast],
+  )
+
+  useEffect(() => {
+    return () => {
+      document.title = 'FormSG'
+    }
+  }, [])
+
+  const isFormNotFound = useMemo(() => {
+    return (
+      error instanceof HttpError &&
+      (error.code === 404 || error.code === 410 || error.code === 403)
+    )
+  }, [error])
+
+  const generateVfnExpiryToast = useCallback(() => {
+    if (vfnToastIdRef.current) {
+      toast.close(vfnToastIdRef.current)
+    }
+    const numVerifiable = data?.form.form_fields.filter((ff) =>
+      get(ff, 'isVerifiable'),
+    ).length
+
+    if (numVerifiable) {
+      vfnToastIdRef.current = toast({
+        duration: null,
+        status: 'warning',
+        isClosable: true,
+        description: simplur`Your verified field[|s] ${[
+          numVerifiable,
+        ]} [has|have] expired. Please verify [the|those] ${[
+          numVerifiable,
+        ]} field[|s] again.`,
+      })
+    }
+  }, [data?.form.form_fields, toast, vfnToastIdRef])
+
+  useTimeout(generateVfnExpiryToast, expiryInMs)
+
+  const isAuthRequired = useMemo(
+    () =>
+      !!data?.form &&
+      data.form.authType !== FormAuthType.NIL &&
+      !data.spcpSession,
+    [data?.form, data?.spcpSession],
+  )
+
+  const { submitEmailModeFormMutation, submitStorageModeFormMutation } =
+    usePreviewFormMutations(formId)
```

Comment:
We should not allow submission on templates - using the preview endpoint may fail if the form does not belong to the user. Just disable altogether imho